tests/test_utils.py

```diff
@@ -1,6 +1,60 @@
 # -*- coding: utf-8 -*-
+import attr
+from typing import Any, Dict
+
 import scrapy
-from scrapy_po.utils import get_callback
+
+from scrapy_po.page_input_providers import PageObjectInputProvider, provides
+from scrapy_po.webpage import Injectable, ItemPage
+
+from scrapy_po import WebPage
+from scrapy_po.utils import (
+    get_callback,
+    is_response_going_to_be_used,
+    DummyResponse,
+)
+
+
+@attr.s(auto_attribs=True)
+class DummyProductResponse:
+
+    data: Dict[str, Any]
+
+
+@provides(DummyProductResponse)
+class DummyProductProvider(PageObjectInputProvider):
+
+    def __init__(self, response: DummyResponse):
+        self.response = response
```

Comment:
It seems this example justifies a refactoring of PageObjectInputProvider: base class here shouldn't require a response, you shouldn't be overriding argument type just to disable base class behavior. Here probably you can get away with an argument-less `__init__`. I was thinking about the same for the autoextract example, but there response was actually needed, to get an url.

Reply:
Done. 

Currently, the source code only injects one dependency, the Response. There was already a comment on the source code telling us to "find out what resources are available". I guess we'll do it later.